lib/rack/attack/store_proxy/redis_cache_store_proxy.rb

```diff
@@ -11,22 +11,27 @@ def self.handle?(store)
         end
 
         def increment(name, amount = 1, options = {})
-          # RedisCacheStore#increment ignores options[:expires_in].
-          #
-          # So in order to workaround this we use RedisCacheStore#write (which sets expiration) to initialize
-          # the counter. After that we continue using the original RedisCacheStore#increment.
-          if options[:expires_in] && !read(name)
-            write(name, amount, options)
-
-            amount
-          else
-            super
+          rescuing do
+            redis.with do |conn|
+              conn.multi do
+                conn.incrby(name, amount)
+                conn.expire(name, options[:expires_in]) if options[:expires_in]
+              end.first
+            end
```

Comment:
This was intentionally done this way in https://github.com/kickstarter/rack-attack/commit/3af7394b6ac6ae97e2cd63eda18daa92a29d0e6b.

My goal at that moment was, and still is, to have Rack::Attack know as less as possible about cache backend details.

Reply:
There's one optimization which can be done without re-implementing which is only `read`ing for activesupport < `v6.0`, because expiration support was added in `v6.0` https://github.com/rails/rails/pull/33254/commits/9d5b02ec5062a23665ec596ef7d3efe4f5abcc27.